mavenembedder-workitem/src/main/java/org/jbpm/process/workitem/mavenembedder/MavenEmbedderWorkItemHandler.java

```
@@ -0,0 +1,148 @@
+                                                     results);
+                    break;
+                case ASYNC:
+                    long workItemId = workItem.getId();
+                    String deploymentId = ((WorkItemImpl) workItem).getDeploymentId() == null ? "" : ((WorkItemImpl) workItem).getDeploymentId();
+                    long processInstanceId = workItem.getProcessInstanceId();
+
+                    new Thread(new Runnable() {
+                        public void run() {
+                            try {
+
+                                Map<String, Object> results = MavenEmbedderUtils.executeMavenGoals(new MavenCli(),
+                                                                                                   RESULTS_VALUES,
+                                                                                                   projectRoot,
+                                                                                                   commandLineOptions,
+                                                                                                   goals,
+                                                                                                   workDir);
+
+                                RuntimeManager manager = RuntimeManagerRegistry.get().getManager(deploymentId);
+                                if (manager != null) {
+                                    RuntimeEngine engine = manager.getRuntimeEngine(ProcessInstanceIdContext.get(processInstanceId));
+
+                                    engine.getKieSession().getWorkItemManager().completeWorkItem(workItemId,
+                                                                                                 results);
+
+                                    manager.disposeRuntimeEngine(engine);
+                                } else {
+                                    kieSession.getWorkItemManager().completeWorkItem(workItemId,
```

remove this else as it will not work anyway - better to throw exception if manager is not found - at least from process execution point of view